packages/dev/core/src/assetContainer.ts

```diff
@@ -1050,4 +1050,62 @@ export class AssetContainer extends AbstractScene {
             }
         });
     }
+
+    /**
+     * @since
+     * Given a root asset, this method will traverse its hierarchy and add it, its children and any materials/skeletons/animation groups to the container.
+     * @param root
+     */
+    public addAllAssetsToContainer(root: Node) {
+        if (!root) {
+            return;
+        }
+
+        const nodesToVisit = new Array<Node>();
+        const visitedNodes = new Set<Node>();
+
+        nodesToVisit.push(root);
+
+        while (nodesToVisit.length > 0) {
+            const nodeToVisit = nodesToVisit.pop()!;
+
+            if (nodeToVisit instanceof Mesh) {
+                if (nodeToVisit.geometry) {
+                    this.geometries.push(nodeToVisit.geometry);

```

Comment:
I think you should check that the geometry has not already been pushed, as several meshes may share the same geometry.

I have the same comment for `materials`, `textures`, `skeletons` and `morphTargetManagers` below.